### DISCUSSION TITLE
Unable to Run code Successfully on Visual Studio

### DISCUSSION THREAD

**Discussion by chokostica** [community/community#66382] [Programming Help]
### Body

[csma-cd-protocol-main.zip](https://github.com/community/community/files/12551421/csma-cd-protocol-main.zip)

I cloned the attached repository on Visual Studio C++. It appears there is no .sln file and hence the code does not run. Someone help with steps how I can run this code successfully on C++.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by seyLu** [comment]
Hi @udlondy 

If you've cloned a C++ repository that doesn't have a Visual Studio solution (.sln) file, you can still build and run the code using Visual Studio by creating a new project manually. Here are the steps to do it:
 
1. Create a new project in Visual Studio
2. Copy the contents into the new project
3. Build and Run as usual
   
You can read more at:
- [How to open C++ proj without .sln in Visual Studio](https://stackoverflow.com/questions/17713037/how-do-i-open-a-project-without-a-sln-file-in-visual-studio)
- [Create a C++ Console App](https://learn.microsoft.com/en-us/cpp/build/vscpp-step-1-create?view=msvc-170)
- [Build and run a C++ Console App](https://learn.microsoft.com/en-us/cpp/build/vscpp-step-2-build?view=msvc-170)

**Reply by chokostica** [reply]
Thank you seyLu.

"Copy the content into the new project"... I dont understand this please. You mean i should open each file on my project, and copy the codes into the new project? Or copy the individual files on my project (I have about 9 files on my project) to the folder of the new project?

**Reply by chokostica** [reply]
![image](https://github.com/community/community/assets/60773335/0b64dcfe-0996-4428-a4e4-2cd5ac888886)
 
I have further tried New > Project From Existing Code, the codes imported successfully. When i ran the program, I got the attached error.

![image](https://github.com/community/community/assets/60773335/5292ba08-24d8-4ed5-adc0-18bec1380079)

**Reply by seyLu** [reply]
Change `SubSystem` to `Console` after following these steps:

![image](https://github.com/community/community/assets/98249191/a05823d5-047a-49d4-9192-b98d4ee1ab20)

source: [MS VisualStudio official docs](https://learn.microsoft.com/en-us/cpp/build/reference/subsystem-specify-subsystem?view=msvc-170#to-set-this-linker-option-in-the-visual-studio-development-environment)

**Comment by aregtech** [comment]
[Here](https://github.com/aregtech/csma-cd-protocol)
